modules/client/src/main/java/org/apache/ignite/internal/client/table/AbstractClientView.java

```diff
@@ -50,6 +63,36 @@ abstract class AbstractClientView<T> implements CriteriaQuerySource<T> {
         this.tbl = tbl;
     }
 
+    /**
+     * Get index mapping from result set to schema.
+     *
+     * @param columns Columns to map.
+     * @param metadata Metadata for query results.
+     * @param startInclusive The first index to cover.
+     * @param endExclusive Index immediately past the last index to cover.
+     * @return Index mapping.
+     */
+    protected static List<Integer> indexMapping(
+            ClientColumn[] columns,
+            int startInclusive,
+            int endExclusive,
+            ResultSetMetadata metadata
+    ) {
+        return Arrays.stream(columns, startInclusive, endExclusive)
+                .map(ClientColumn::name)
+                .map(IgniteNameUtils::quoteIfNeeded)
```

Comment:
Added `IgniteNameUtilsTest#quoteIfNeeded`

Reply:
Column name in `ClientColumn#name`, `Column#name` stored without quotes.

Reply:
dangerous it looks because I don't uderstand all cases when and how it will works. It need to cover by integration test with different names - quoted, quted contains quoted in name, and so one..